sourcegraph/jetbrains
# fix/Webview: Make Cody: Restart Cody close Webview panels and restore Webview views

Fixes CODY-3546.

Hook the "stop agent" method to clean up webviews.

Some tidy ups:

- Move webviews into their own package.
- Break large webview files apart, and organize them by layer.
- Add a README.md describing this structure.
- Be explicit about UNIX-style line endings with Spotless.

## Test plan

1. Open some chats, chat in the sidebar
2. Shift-shift, Cody: Restart Cody action
3. Verify that the chat panels close and the sidebar reloads

## Changelog

The webview-based chat now works after restarting Cody.
